gordo_components/dataset/datasets.py

```diff
@@ -64,20 +74,31 @@ def __init__(
                 f"information"
             )
 
-    def get_data(self) -> Tuple[pd.DataFrame, None]:
-        dataframes = self.data_provider.load_series(
-            from_ts=self.from_ts, to_ts=self.to_ts, tag_list=self.tag_list
+    def get_data(self) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
+
+        series_iter: Iterable[pd.Series] = self.data_provider.load_series(
+            from_ts=self.from_ts,
+            to_ts=self.to_ts,
+            tag_list=list(set(self.tag_list + self.target_tag_list)),
+        )
+        data: pd.DataFrame = self.join_timeseries(
+            series_iter, self.from_ts, self.to_ts, self.resolution
         )
-        X = self.join_timeseries(dataframes, self.from_ts, self.to_ts, self.resolution)
-        y = None
         if self.row_filter:
-            X = pandas_filter_rows(X, self.row_filter)
-        logger.info(f"First five rows of the filtered dataset are {X.head()}")
+            data = pandas_filter_rows(data, self.row_filter)
+
+        x_tag_names = [tag.name for tag in self.tag_list]
+        y_tag_names = [tag.name for tag in self.target_tag_list]
+
+        X = data[x_tag_names]
```

Comment:
So this is the thing which ensures that you get the tags out again in the expected order (i.e. in the same order as they were in the `tag_list`? Can you maybe add a test which verifies this, i.e. one where the sensors are listed as `["Sensor 2", "Sensor 1"]`, and ensure that they come out with the first columns being `Sensor 2`

Reply:
Very good, as of now, it will return them in the same order, but definitely good to make a test to verify it stays that way. :100:

Reply:
:boom: :checkered_flag: